modules/gapi/src/backends/common/gbackend.hpp

```diff
@@ -58,32 +58,37 @@ namespace gimpl {
     struct Data;
     struct RcDesc;
 
-    struct GAPI_EXPORTS RMatMediaAdapterBGR final: public cv::RMat::Adapter
+    struct GAPI_EXPORTS RMatMediaFrameAdapter final: public cv::RMat::Adapter
     {
-        explicit RMatMediaAdapterBGR(const cv::MediaFrame& frame) : m_frame(frame) { };
+        explicit RMatMediaFrameAdapter(const cv::MediaFrame& frame,
+            std::function<cv::Mat(const GFrameDesc&, const cv::MediaFrame::View&)> frameViewToMat) :
+            m_frame(frame),
+            m_frameViewToMat(frameViewToMat)
+        {
+            const auto& d = m_frame.desc();
+            const auto& fv = m_frame.access(cv::MediaFrame::Access::R);
```

Comment:
I'd delay this to the actual `access` - why do we map MediaFrame's data immediately?

Is there a performance benefit in our case?

Reply:
Ah you only do it to obtain meta early -- then I'd move it to that point.

Reply:
Also, if you'll do this, please take care of concurrency.

Reply:
Removed this part.
Implemented retrieving of meta as discussed internally.